subprojects/ide-native/src/main/java/org/gradle/ide/xcode/internal/xcodeproj/PBXProject.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.ide.xcode.internal.xcodeproj;
+
+import com.dd.plist.NSDictionary;
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Ordering;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * The root object representing the project itself.
+ */
+public class PBXProject extends PBXContainer {
+    private final PBXGroup mainGroup;
+    private final List<PBXTarget> targets;
+    private final XCConfigurationList buildConfigurationList;
+    private final String compatibilityVersion;
+    private String name;
+
+    public PBXProject(String name) {
```

Comment:
It's kind of weird that you can pass in a parameter but later change it with a setter method. I would have expected for the field to be `final`.